modules/gateway/conn.go

```diff
@@ -24,13 +24,13 @@ func (pc peerConn) RPCAddr() modules.NetAddress {
 // communication protocol.
 func (g *Gateway) dial(addr modules.NetAddress) (net.Conn, error) {
 	dialer := &net.Dialer{
-		Timeout: dialTimeout,
-		Cancel:  g.threads.StopChan(),
+		Cancel:   g.threads.StopChan(),
+		Deadline: time.Now().Add(connStdDeadline),
+		Timeout:  dialTimeout,
 	}
 	conn, err := dialer.Dial("tcp", string(addr))
 	if err != nil {
 		return nil, err
 	}
-	conn.SetDeadline(time.Now().Add(connStdDeadline))
```

Comment:
Making this changed fixed a bug where the gateway's were having a substantial amount of i/o timeouts. I'm not sure where I read that I should be doing that, but I think it was in google-gropus. I think it said that `Dial` we reuse conns, and so you want to set the timeout in the dialer instead of on the conn itself.

Now I'm second guessing myself. John can back me up that it solved a bunch of i/o timeout warnings in the gateway.